Makefile

```diff
@@ -0,0 +1,69 @@
+UBUNTU_VERSION ?= bionic
+
+UBUNTU_VENTI_GOVERSIONS ?= 1.8 1.9 1.10.3 1.10.7 1.10.8 1.11.4 1.11.5
+
+REGISTRY ?= quay.io/gravitational
+
+DOCKER_COMMON_OPTS = --rm --privileged \
+	-e DEBIAN_FRONTEND=noninteractive \
+	-e http_proxy=$(http_proxy) \
+	-e UBUNTU_VERSION=$(UBUNTU_VERSION) \
+	-v $(shell pwd):/build:ro
+
+.PHONY: all
+all: images
+
+.PHONY: images
+images: ubuntu-tall ubuntu-grande ubuntu-venti ubuntu-venti-go
+
+.PHONY: ubuntu-tall
+ubuntu-tall:
+	-docker rmi ubuntu-tall:$(UBUNTU_VERSION)
+	docker run $(DOCKER_COMMON_OPTS) \
+		ubuntu:$(UBUNTU_VERSION) bash /build/tall/build.sh > tall.tar
+	docker import \
+		tall.tar ubuntu-tall:$(UBUNTU_VERSION)
+
+.PHONY: ubuntu-grande
+ubuntu-grande:
+	-docker rmi ubuntu-grande:$(UBUNTU_VERSION)
+	docker run $(DOCKER_COMMON_OPTS) \
+		ubuntu:$(UBUNTU_VERSION) bash /build/grande/build.sh > grande.tar
+	docker import \
+		--change 'ENV DEBIAN_FRONTEND noninteractive' \
+		grande.tar ubuntu-grande:$(UBUNTU_VERSION)
+
+.PHONY: ubuntu-venti
+ubuntu-venti:
+	-docker rmi ubuntu-venti:$(UBUNTU_VERSION)
+	docker run $(DOCKER_COMMON_OPTS) \
+		ubuntu:$(UBUNTU_VERSION) bash /build/venti/build.sh > venti.tar
+	docker import \
+		--change 'ENV DEBIAN_FRONTEND noninteractive' \
+		venti.tar ubuntu-venti:$(UBUNTU_VERSION)
+
+.PHONY: ubuntu-venti-go
+ubuntu-venti-go:
+	for goversion in $(UBUNTU_VENTI_GOVERSIONS) ; do \
+		docker rmi ubuntu-venti:go$$goversion-$(UBUNTU_VERSION) || true ; \
+		docker build --build-arg GOVERSION=$$goversion -t ubuntu-venti:go$$goversion-$(UBUNTU_VERSION) venti ; \
+	done
+
+.PHONY: syntax-check
+syntax-check:
+	find . -name '*.sh' | xargs shellcheck
+
+.PHONY: push
+push:
+	for goversion in $(UBUNTU_VENTI_GOVERSIONS); do \
+		docker tag ubuntu-venti:go$$goversion-$(UBUNTU_VERSION) $(REGISTRY)/ubuntu-venti:go$$goversion-$(UBUNTU_VERSION) && \
+		docker push $(REGISTRY)/ubuntu-venti:go$$goversion-$(UBUNTU_VERSION) ; \
+	done
+	docker tag ubuntu-venti:$(UBUNTU_VERSION) $(REGISTRY)/ubuntu-venti:$(UBUNTU_VERSION)
+	for version in 0.0.2 $(UBUNTU_VERSION); do \
```

Comment:
0.0.2, for now, is used in debian-based images. You have to use another tag.

Reply:
Can you explain how this works and what it's used for?

Reply:
I am not sure I followed what to explain? We use quya.io/gravitational/debian-XXXX images with specific tags like `stretch`, `0.0.1` in our development process. You someone using `0.0.2` tag and expecting there Debian with adding this change you switch Debian image with this tag to Ubuntu image and it fill be unexpected behavior for end users(even if end users only our company projects).

Reply:
I think that's part of the point, though - we should be able to change all our Debian-based images to Ubuntu-based images because they're (theoretically) compatible.

Also, if someone pulls `quay.io/gravitational/docker-debian:0.0.2` they're still going to get Debian - this repo's image would be `quay.io/gravitational/docker-ubuntu:0.0.2`.

Reply:
Important word here is `theoretically`. Changing root image is like an changing API without warning end users about it. We can do that, but if someones get unexpected result the image will be the last thing which end will think about because he expects no changes in EXISTING tag. 
Plus, tags in docker are like git commits. With special keywords like `latest`, `stretch` and etc which points to specific `branch` of images. 0.0.2 is a static tag and should not be changed. We use 0.0.1 tag a bit wrong in that case, but it is because we have to rebuild every image every 3 days for security packages update.

Reply:
This is why we're making this a fresh repo called `docker-ubuntu` - nothing is going to change about `docker-debian:0.0.2`. It will still be a Debian image (until anyone using it deliberately changes their `Dockerfile` to say `docker-ubuntu:0.0.2`, for example)

I think what I don't quite understand is what's special about `0.0.2` here? Why would anyone expect tag `0.0.2` in a repo called `docker-ubuntu` to be a Debian image and not an Ubuntu image?